portality/static/doaj/doajArticles.xsd

```diff
@@ -130,15 +131,13 @@
      </xs:simpleContent>
     </xs:complexType>
    </xs:element>
- 
-   <xs:element name="fullTextUrl">
-    <xs:complexType> 
-     <xs:simpleContent>
-      <xs:extension base="xs:anyURI">
-       <xs:attribute name="format" />  
-      </xs:extension>
-     </xs:simpleContent>
-    </xs:complexType>
+
+         <xs:element name="fullTextUrl">
+    <xs:simpleType>
+     <xs:restriction base="xs:anyURI">
+      <xs:pattern value="https?://([^/:]+\.[\p{L}]{2,10}|([0-9]{1,3}\.){3}[0-9]{1,3})(:[0-9]+)?(/.*)?"/>
```

Comment:
This looks like a well designed regex, but we might be being overly prescriptive for this use case.

That said, I notice it's very similar to the URL regex we use in https://github.com/DOAJ/doaj/blob/develop/portality/formcontext/fields.py#L5 - and in the validator here: https://github.com/DOAJ/doaj/blob/develop/portality/formcontext/validate.py#L187

The fulltext field in the article metadata form, however, uses WtForms' `validators.url`:
```python
regex = r'^[a-z]+://(?P<host>[^/:]+)(?P<port>:[0-9]+)?(?P<path>\/.*)?$'
```

Interestingly, none of the existing examples use the syntax to allow unicode in URLs, but I'm not sure if that's taken care of by Python elsewhere.

Reply:
It's also full of unecessary capture groups `()` - but that does keep it fairly close to our existing examples.

I'm torn between completely stripping this down to something like `^https?://.+` and letting the `anyURI` type handle the rest of the validation, or leaving it as it's written here and trust it to do its job since the tests pass.

Reply:
The regex was intentionally done (as discussed with @richard-jones ) as an equivalent of the one from https://github.com/DOAJ/doaj/blob/develop/portality/formcontext/fields.py#L5

Urls from the forms are later checked if they start with the http(s) via API - so I added that check to the regex to ensure it works the same as API. The last change is using \p{L] instead of [a-zA-Z] that ensures accepting also non-unicode characters and is xml regex feature as described here:   https://www.regular-expressions.info/unicode.html in "Unicode Categories" section. I can check if python itself allowes unicode in regex as we should probably change the regex from formcontext/fields to one accepting them as well.

Reply:
I've been bashing at this regex a bit, and it seems to work well, with only two edge cases that we might need to take into account.  Neither of the following validate, but I think that they would be considered valid:

http://example.com?q=500&whatever=true
http://example.com#this

If you insert "/" after ".com" then these do validate.  If you put them in a browser, the browser redirects to the version with the "/" inserted, which would suggest that is the preferred form.  I'm cautious about introducing something that would possibly throw an error on an otherwise usable URL though.